pkg/pillar/types/smarttypes.go

```diff
@@ -1,65 +1,88 @@
 /*
- * Copyright (c) 2020. Zededa, Inc.
+ * Copyright (c) 2022. Zededa, Inc.
  * SPDX-License-Identifier: Apache-2.0
  */
 
 package types
 
-//SmartData holds SMART details of the device
-type SmartData struct {
-	SmartCtl        SmartCtl       `json:"smartctl"`
-	Device          Device         `json:"device"`
-	SmartAttribute  SmartAttribute `json:"ata_smart_attributes"`
-	PowerOnTime     PowerOnTime    `json:"power_on_time"`
-	PowerCycleCount int64          `json:"power_cycle_count"`
-	Temperature     Temperature    `json:"temperature"`
-	RawData         string         `json:"raw_data"`
+//DeviceSmartInfo holds SMART details of the device
+type DeviceSmartInfo struct {
+	Smartctl           cmdSmartctl     `json:"smartctl"`
+	Device             device          `json:"device"`
+	ModelName          string          `json:"model_name"`
+	SerialNumber       string          `json:"serial_number"`
+	Wwn                smWWN           `json:"wwn"`
+	AtaSmartAttributes SmartAttributes `json:"ata_smart_attributes"`
+	Temperature        Temperature     `json:"temperature"`
+	PowerOnTime        PowerOnTime     `json:"power_on_time"`
+	PowerCycleCount    int64           `json:"power_cycle_count"`
 }
 
-//Device holds device details
-type Device struct {
+//DisksList holds list with disks
+type DisksList struct {
+	Smartctl cmdSmartctl `json:"smartctl"`
+	Devices  []device    `json:"devices"`
+}
+
+type errMessages struct {
+	String   string `json:"string"`
+	Severity string `json:"severity"`
+}
+
+//cmdSmartctl query result.
+type cmdSmartctl struct {
+	Argv       []string      `json:"argv"`
+	Messages   []errMessages `json:"messages"`
+	ExitStatus int           `json:"exit_status"`
+}
+
+//device holds device details
+type device struct {
```

Comment:
I see you unexported this type by changing it from Device to device, and that some of the new types are exported and others are not. How did you determine what to export and what not to export?

Reply:
I decided to completely remove this unused code, since it is currently not used and its further use is not planned. Now s.m.a.r.t information is collected in a different way and we no longer need most of the structures for processing JSON.

Only the DeviceSmartInfo structure and the PowerCycleCount reading part, which is used in nodeagent and reportinfo, remained, but if this is to be redone, then it will be a separate PR.